lib/esestream.cpp

```diff
@@ -90,15 +109,23 @@ ESEStream::prepareFrame (ESEBuffer buffer, uint32_t start,
 }
 
 ESEPacket *
-ESEStream::prepareCurrentPacket (uint64_t pts, uint64_t dts, uint64_t duration)
+ESEStream::prepareNextPacket (uint64_t pts, uint64_t dts, uint64_t duration)
 {
-  m_currentPacket = new ESEPacket ();
-  m_currentPacket->data = m_currentFrame.data ();
-  m_currentPacket->data_size = m_currentFrame.size ();
-  m_currentPacket->pts = pts;
-  m_currentPacket->dts = dts;
-  m_currentPacket->duration = duration;
+  m_nextPacket = new ESEPacket ();
+  m_nextPacket->data = reinterpret_cast<std::uint8_t*>(std::malloc(m_currentFrame.size ()));
```

Comment:
```suggestion
  m_nextPacket->data = static_cast<std::uint8_t*>(std::malloc(m_currentFrame.size ()));
```

reinterpret_cast use run time logic to verify if the casting is possible (for example between object instances) and that logic is not required here